yoda/yoda.h

```diff
@@ -23,126 +23,52 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-
-template <typename T_ENTRY>
-struct KeyAlreadyExistsException : KeyAlreadyExistsCoverException {
-  const T_ENTRY entry;
-  explicit KeyAlreadyExistsException(const T_ENTRY& entry) : entry(entry) {}
-};
-
-struct EntryShouldExistCoverException : bricks::Exception {};
-
-template <typename T_ENTRY>
-struct EntryShouldExistException : EntryShouldExistCoverException {
-  const T_ENTRY entry;
-  explicit EntryShouldExistException(const T_ENTRY& entry) : entry(entry) {}
-};
-
-// Base structures for user entry types.
-struct AllowNonThrowingGet {
-  constexpr static bool allow_nonthrowing_get = true;
-};
-
-enum NullEntryTypeHelper { NullEntry };
-struct Nullable {
-  const bool exists;
-  Nullable() : exists(true) {}
-  Nullable(NullEntryTypeHelper) : exists(false) {}
-};
-
-struct Deletable {};  // User promises to serialize Nullable::exists;
-
-// Universal null entry creation.
-template <typename T_ENTRY, bool IS_NULLABLE>
-struct CreateNullEntryImpl {
-  static T_ENTRY get() {
-    // This function shoud NEVER be called.
-    // If you see this exception, something went terribly wrong.
-    BRICKS_THROW(NonThrowingGetPrerequisitesNotMetException(
-        "Creating NullEntry requested for the entry type, which is not derived from Nullable."));
-  }
-};
-
-template <typename T_ENTRY>
-struct CreateNullEntryImpl<T_ENTRY, true> {
-  static T_ENTRY get() {
-    T_ENTRY e(NullEntry);
-    return e;
-  }
-};
-
-template <typename T_ENTRY, bool IS_NULLABLE>
-T_ENTRY CreateNullEntry() {
-  return CreateNullEntryImpl<T_ENTRY, IS_NULLABLE>::get();
-}
-
-// API calls, `[Async]{Get/Add/Update/Delete}(...)`.
+// Yoda is a simple to use, machine-learning-friendly key-value storage atop a Sherlock stream.
 //
-// C++ API supports three syntaxes:
+// The C++ usecase is to instantiate `Yoda<MyEntryType> API("stream_name")`.
+// The API supports CRUD-like operations, presently `Get` and `Add`.
 //
-//   1) Asynchronous, using std::{promise/future}.
-//      To use the async/await paradigm.
-//      `AsyncAdd(entry)` and `AsyncGet(key)`, both return a future.
+// The API offers three ways to invoke itself:
 //
-//   2) Asynchronous, using std::function as callbacks.
-//      Best for HTTP endpoints, with the Request object passed into these callbacks.
-//      `AsyncAdd(entry, on_success, on_failure)` and `AsyncGet(key, on_success, on_failure)`, both return void.
+//   1) Asynchronous, using std::{promise/future}, that utilize the the async/await paradigm.
+//      For instance, `AsyncAdd(entry)` and `AsyncGet(key)`, which both return a future.
 //
-//   3) Synchronous. `Get(key)` and `Add(entry)`.
+//   2) Asynchronous, using std::function as callbacks. Best for HTTP endpoints and other thread-based usecases.
+//      For instance, `AsyncAdd(entry, on_success, on_failure)` and `AsyncGet(key, on_success, on_failure)`,
+//      which both return `void` immediately and invoke the callback as the result is ready, from a separate
+//      thread.
```

Comment:
I'm not sure that I understand the sentence right, but we've just removed std::async()'ed callbacks, haven't we?